src/sparse_blas/backends/cusparse/cusparse_global_handle.hpp

```diff
@@ -0,0 +1,63 @@
+/***************************************************************************
+*  Copyright (C) Codeplay Software Limited
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+*  For your convenience, a copy of the License has been included in this
+*  repository.
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*
+**************************************************************************/
+
+#ifndef _ONEMKL_SPARSE_BLAS_BACKENDS_CUSPARSE_GLOBAL_HANDLE_HPP_
+#define _ONEMKL_SPARSE_BLAS_BACKENDS_CUSPARSE_GLOBAL_HANDLE_HPP_
+
+/**
+ * @file Similar to blas_handle.hpp
+ * Provides a map from a ur_context_handle_t (or equivalent) to a cusparseHandle_t.
+ * @see cusparse_scope_handle.hpp
+*/
```

Comment:
Wasn't it that the whole point of having `sycl::queue` be a part of all sparse BLAS APIs in the oneAPI specification was to prevent needing a map like this, and that the `queue` would somehow be used to track `cusparseHandle_t`?

Reply:
Unfortunately no, it is not enough to have the `sycl::queue` to avoid this map. The blas domain also always has the queue and need this map.
The queue is needed for oneMKL runtime dispatch to choose which backend can be used with a given device.
The `sycl::queue` alone can't store this. The suggested solution to avoid this map was to introduce a global object that the user must initialize with the `sycl::queue` and give this object to every function instead of the `sycl::queue`. We decided against it as the other backends don't do this.

Reply:
I see. Thanks for the clarification.